### PR TITLE
Add testing for Notes

### PR DESCRIPTION
**Pull request by rtshkmr**
## Description

Improves existing tests by using test utils and cleans up todo comments in code. 

Fixes #

## Testing

NIL

## Remarks

This is a follow up to the mysterious merging of #203 earlier in the day.

### CODE REVIEW

File: src/main/java/seedu/address/model/note/TagNoteMap.java [hunk 1]
```diff
@@ -136,11 +136,7 @@ public void deleteNote(Note note) {
      * @param newTags The tags to associate with a particular note.
```
Comment (qwoprocks) [inline src/main/java/seedu/address/model/note/TagNoteMap.java:136]: Need to update javadoc for this method to match the method behavior

File: src/test/java/seedu/address/logic/parser/ClientNoteAddCommandParserTest.java [hunk 2]
```diff
@@ -72,7 +71,7 @@ public void parse_hasIndexHasNote_equalsExpected() {
             expectedClientNote.setTags(untaggedTags);
             ClientNoteAddCommand expected = new ClientNoteAddCommand(Index.fromOneBased(1), expectedClientNote);
             ClientNoteAddCommand actual = parser.parse(HAS_INDEX_HAS_NOTE);
```
Comment (qwoprocks) [inline src/test/java/seedu/address/logic/parser/ClientNoteAddCommandParserTest.java]: This variable isn't being used

**Review by qwoprocks — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/note/TagNoteMap.java (hunk 1 above)
Comment (rtshkmr) [reply src/main/java/seedu/address/model/note/TagNoteMap.java:136]: resolved, added mention of `uniqueTagMap`

**Review by qwoprocks — APPROVED** [review]
LGTM

**Pull request merged**